markdownify/__init__.py

```diff
@@ -416,8 +463,9 @@ def convert_list(self, el, text, convert_as_inline):
 
         nested = False
         before_paragraph = False
-        if el.next_sibling and el.next_sibling.name not in ['ul', 'ol']:
-            before_paragraph = True
+        if next_sibling := _next_content_sibling(el):
```

Comment:
Using the walrus operator [excludes every python version below 3.8](https://peps.python.org/pep-0572/). This can be a choice, but currently this package [is listed as 2.5/6/7 and 3.6/7/8 compatible](https://github.com/matthewwithanm/python-markdownify/blob/ae0597d80cb57983e876d13fdd44a7728abcbe26/pyproject.toml#L11) in pip.

Reply:
I thought because `tox.ini` had a `py38` reference that we might be Python 3.8 and above. But I hadn't seen the entries in `pyproject.toml` - thanks! I have rewritten to avoid the walrus operator.